lib/portage/util/env_update.py

```diff
@@ -363,19 +364,24 @@ def _env_update(makelinks, target_root, prev_mtimes, contents, env, writemsg_lev
             writemsg_level(
                 _(">>> Regenerating %setc/ld.so.cache...\n") % (target_root,)
             )
-            os.system(f"cd / ; {ldconfig} -X -r '{target_root}'")
+            ret = os.system(f"cd / ; {ldconfig} -X -r '{target_root}'")
         elif ostype in ("FreeBSD", "DragonFly"):
             writemsg_level(
                 _(">>> Regenerating %svar/run/ld-elf.so.hints...\n") % target_root
             )
-            os.system(
+            ret = os.system(
                 (
                     "cd / ; %s -elf -i "
                     + "-f '%svar/run/ld-elf.so.hints' '%setc/ld.so.conf'"
                 )
                 % (ldconfig, target_root, target_root)
             )
 
+        ret = os.waitstatus_to_exitcode(ret)
```

Comment:
This function was added in Python 3.9, but we at least pretend to support 3.7 and 3.8.

Reply:
To support older Pythons, I think some combination of `os.WIFSIGNALED`, `os.WIFEXITED`, `os.WEXITSTATUS`, and `os.WTERMSIG` will be needed.

https://docs.python.org/3.8/library/os.html#os.WIFSIGNALED

Reply:
Something like this should work.
```suggestion
        if os.WIFEXITED(ret):
            ret = os.WEXITSTATUS(ret)
            writemsg(f"!!! ldconfig failed with exit status {ret}\n", noiselevel=-1)
        elif os.WIFSIGNALED(ret):
            ret = os.WTERMSIG(ret)
            writemsg(f"!!! ldconfig was killed with signal {ret}\n", noiselevel=-1)
```

Reply:
Ah, thanks. Maybe it's time to stop pretending though?

Reply:
Python 3.7 is officially end-of-life now, so no objection from me if you want to drop it.

```suggestion
        ret = os.waitstatus_to_exitcode(ret)
        if ret > 0:
            writemsg(f"!!! ldconfig failed with exit status {ret}\n", noiselevel=-1)
        if ret < 0:
            writemsg(f"!!! ldconfig was killed with signal {-ret}\n", noiselevel=-1)
```

Reply:
Thanks (on both counts).